store/tikv/2pc_fail_test.go

```diff
@@ -100,3 +116,300 @@ func (s *testCommitterSuite) TestFailCommitTimeout(c *C) {
+	s.mustCommit(c, map[string]string{
+		"a": "a0",
+		"b": "b0",
+	})
+
+	txn := s.begin(c)
+	err := txn.Set([]byte("a"), []byte("a1"))
+	c.Assert(err, IsNil)
+	err = txn.Set([]byte("b"), []byte("b1"))
+	c.Assert(err, IsNil)
+
+	gofail.Enable("github.com/pingcap/tidb/store/tikv/tikvStoreSendReqResult", `2*off->1*return("timeout")`)
+	err = txn.Commit(context.Background())
+	c.Assert(err, NotNil)
+	gofail.Disable("github.com/pingcap/tidb/store/tikv/tikvStoreSendReqResult")
+
+	txnCheck := s.begin(c)
+	var value []byte
+	value, err = txnCheck.Get([]byte("a"))
+	c.Assert(err, IsNil)
+	c.Assert(value, BytesEquals, []byte("a0"))
+
+	value, err = txnCheck.Get([]byte("b"))
+	c.Assert(err, IsNil)
+	c.Assert(value, BytesEquals, []byte("b0"))
+}
+
+func (s *testCommitterSuite) TestFailGetTimeOut(c *C) {
+	s.mustCommit(c, map[string]string{
+		"a": "a0",
+	})
+
+	txn := s.begin(c)
+	gofail.Enable("github.com/pingcap/tidb/store/tikv/tikvStoreSendReqResult", `1*return("timeout")`)
+	value, err := txn.Get([]byte("a"))
+	c.Assert(err, NotNil)
+	c.Assert(value, IsNil)
+	gofail.Disable("github.com/pingcap/tidb/store/tikv/tikvStoreSendReqResult")
+}
+
+func (s *testCommitterSuite) TestFailCommitSecondaryKeyTimeOut(c *C) {
+	s.mustCommit(c, map[string]string{
+		"a": "a0",
+		"b": "b0",
+	})
+
+	txn := s.begin(c)
+	err := txn.Set([]byte("a"), []byte("a1"))
+	c.Assert(err, IsNil)
+	err = txn.Set([]byte("b"), []byte("b1"))
+	c.Assert(err, IsNil)
+
+	gofail.Enable("github.com/pingcap/tidb/store/tikv/tikvStoreSendReqResult", `3*off->1*return("timeout")`)
+	err = txn.Commit(context.Background())
+	c.Assert(err, IsNil)
+	gofail.Disable("github.com/pingcap/tidb/store/tikv/tikvStoreSendReqResult")
+
+	txnCheck := s.begin(c)
+	var value []byte
+	value, err = txnCheck.Get([]byte("a"))
+	c.Assert(err, IsNil)
+	c.Assert(value, BytesEquals, []byte("a1"))
+
+	value, err = txnCheck.Get([]byte("b"))
+	c.Assert(err, IsNil)
+	c.Assert(value, BytesEquals, []byte("b1"))
+}
+
+func (s *testCommitterSuite) TestFailLockKeysTimeOut(c *C) {
+	s.mustCommit(c, map[string]string{"a": "a0"})
+
+	txn1 := s.begin(c)
+	err := txn1.LockKeys([]byte("a"))
+	c.Assert(err, IsNil)
+
+	txn2 := s.begin(c)
+	err = txn2.Set([]byte("a"), []byte("a1"))
+	c.Assert(err, IsNil)
+
+	gofail.Enable("github.com/pingcap/tidb/store/mockstore/mocktikv/rpcCommitResult", `return("timeout")`)
+	err = txn1.Commit(context.Background())
+	c.Assert(err, NotNil)
+	gofail.Disable("github.com/pingcap/tidb/store/mockstore/mocktikv/rpcCommitResult")
+
+	err = txn2.Commit(context.Background())
+	c.Assert(err, IsNil)
+
+	txnCheck := s.begin(c)
+	var value []byte
+	value, err = txnCheck.Get([]byte("a"))
+	c.Assert(err, IsNil)
+	c.Assert(value, BytesEquals, []byte("a1"))
+}
+
+func (s *testCommitterSuite) TestTwoPhaseCommitActionString(c *C) {
+	action := twoPhaseCommitAction(0)
+	c.Assert(action.String(), Equals, "unknown")
+}
+
+/*func (s *testCommitterSuite) TestFailNewTwoPhaseCommitter(c *C) {
```

Comment:
Why comment this function?

Reply:
this case will do harm to other test cases. I will enable it in the future.

Reply:
So add comment for it and use
```
c.Skip("reason")
```

Reply:
ok